lib/matplotlib/tests/test_axes.py

```diff
@@ -4863,6 +4863,25 @@ def test_pandas_indexing_hist():
     axes.hist(ser_2)
 
 
+def test_pandas_bar_align_center():
+    # Tests fix for issue 8767
+    pd = pytest.importorskip('pandas')
+
+    df = pd.DataFrame({'col1': [1] + [2],
+                       'col2': [1]*2,
+                       'col3': [1]*2})
+
+    fig, axArr = plt.subplots(2)
+
+    curr = 1
+    for ax in axArr:
```

Comment:
I could not recreate the error found inn [8767](https://github.com/matplotlib/matplotlib/issues/8767) without the 2 subplots.